crates/matrix-sdk/src/room/timeline/event_item.rs

```diff
@@ -159,101 +158,140 @@ impl EventTimelineItem {
-        Self { reactions, ..self.clone() }
+impl fmt::Debug for LocalEventTimelineItem {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("LocalEventTimelineItem")
+            .field("transaction_id", &self.transaction_id)
+            .field("event_id", &self.event_id)
+            .field("sender", &self.sender)
+            .field("timestamp", &self.timestamp)
+            .field("content", &self.content)
+            .field("encryption_info", &self.encryption_info)
+            // skip raw, too noisy
+            .finish_non_exhaustive()
     }
 }
 
-/// A unique identifier for a timeline item.
-///
-/// This identifier is used to find the item in the timeline in order to update
-/// its state.
-///
-/// When an event is created locally, the timeline reflects this with an item
-/// that has a [`TransactionId`](Self::TransactionId) key. Once the server has
-/// acknowledged the event and given it an ID, that item's key is replaced by
-/// [`EventId`](Self::EventId) containing the new ID.
-///
-/// When an event related to the original event whose ID is stored in a
-/// [`TimelineKey`] is received, the key is left untouched, but other parts of
-/// the timeline item may be updated. Thus, the current data model is only able
-/// to handle relations that reference the initial event that resulted in a
-/// timeline item being created, not other related events. At the time of
-/// writing, there is no relation that is meant to refer to other events that
-/// only exist for their relation (e.g. edits, replies).
-#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
-pub enum TimelineKey {
-    /// Transaction ID, for an event that was created locally and hasn't been
-    /// acknowledged by the server yet.
-    TransactionId {
-        /// The transaction ID.
-        txn_id: OwnedTransactionId,
-        /// The event ID received from the server in the event-sending response
-        /// (not the sync response).
-        event_id: Option<OwnedEventId>,
-    },
-    /// Event ID, for an event that is synced with the server.
-    EventId(OwnedEventId),
+#[derive(Clone)]
+pub struct RemoteEventTimelineItem {
+    /// The event ID.
+    pub event_id: OwnedEventId,
+    /// The sender of the event.
+    pub sender: OwnedUserId,
+    /// The sender's profile of the event.
+    pub sender_profile: Profile,
+    /// The timestamp of the event.
+    pub timestamp: MilliSecondsSinceUnixEpoch,
+    /// The content of the event.
+    pub content: TimelineItemContent,
+    /// All bundled reactions about the event.
+    pub reactions: BundledReactions,
+    /// Whether the event has been sent by the the logged-in user themselves.
+    pub is_own: bool,
+    /// Encryption information.
+    pub encryption_info: Option<EncryptionInfo>,
+    // FIXME: Expose the raw JSON of aggregated events somehow
+    pub raw: Option<Raw<AnySyncTimelineEvent>>,
```

Comment:
This should always be `Some` as it's the source of this item.

Reply:
Removed, thanks!